src/widget/controlwidgetconnection.cpp

```diff
@@ -97,40 +97,67 @@ void ControlParameterWidgetConnection::setControlParameterUp(double v) {
 }
 
 ControlWidgetPropertyConnection::ControlWidgetPropertyConnection(
-        WBaseWidget* pBaseWidget, const ConfigKey& key,
-        ValueTransformer* pTransformer, const QString& propertyName)
+        WBaseWidget* pBaseWidget,
+        const ConfigKey& key,
+        ValueTransformer* pTransformer,
+        const QString& propertyName)
         : ControlWidgetConnection(pBaseWidget, key, pTransformer),
-          m_propertyName(propertyName.toLatin1()) {
+          m_propertyName(propertyName) {
+    QWidget* pWidget = pBaseWidget->toQWidget();
+    VERIFY_OR_DEBUG_ASSERT(pWidget) {
+        return;
+    }
+    const QMetaObject* meta = pWidget->metaObject();
+    VERIFY_OR_DEBUG_ASSERT(!propertyName.isEmpty() && meta) {
+        return;
+    }
+    int id = meta->indexOfProperty(propertyName.toLatin1().constData());
+    VERIFY_OR_DEBUG_ASSERT(id >= 0) {
+        return;
+    }
+    m_property = meta->property(id);
+    // skipNoOp = false, during the initial update to synchronize the property in all the sub widgets
     slotControlValueChanged(m_pControl->get());
 }
 
 QString ControlWidgetPropertyConnection::toDebugString() const {
     const ConfigKey& key = getKey();
-    return QString("%1,%2 Parameter: %3 Property: %4 Value: %5").arg(
-            key.group,
-            key.item,
-            QString::number(m_pControl->getParameter()),
-            m_propertyName,
-            m_pWidget->toQWidget()->property(m_propertyName.constData()).toString());
+    return QString("%1,%2 Parameter: %3 Property: %4 Value: %5")
+            .arg(key.group,
+                    key.item,
+                    QString::number(m_pControl->getParameter()),
+                    m_propertyName,
+                    m_property.read(m_pWidget->toQWidget()).toString());
 }
 
 void ControlWidgetPropertyConnection::slotControlValueChanged(double v) {
-    QVariant parameter;
-    QWidget* pWidget = m_pWidget->toQWidget();
-    QVariant property = pWidget->property(m_propertyName.constData());
-    if (property.type() == QVariant::Bool) {
-        parameter = getControlParameterForValue(v) > 0;
+    double parameter = getControlParameterForValue(v);
+    QVariant vParameter;
+    if (m_property.type() == QVariant::Bool) {
+        vParameter = (parameter > 0);
     } else {
-        parameter = getControlParameterForValue(v);
+        vParameter = parameter;
+    }
+    vParameter.convert(m_property.type());
```

Comment:
The conversion might fail according to the function signature but is never expected to fail -> debug assertion